languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -10069,6 +10069,36 @@ though, and has since been slightly modified)-->
     </rulegroup>
   </category>
   <category id="CAT_GRAMMAIRE" name="Grammaire">
+      <rulegroup id="ETRE_TOUTES_OU_TOUS_V_PPA" name="être toutes ou tous v ppa" default="temp_off">
+          <rule>
+              <pattern>
+                  <token postag="R.* pers.*" postag_regexp="yes"></token>
+                  <token postag="V.* etre.*" postag_regexp="yes"></token>
+                  <marker>
+                      <token regexp="yes">toute|tous</token>
+                  </marker>
+                  <token postag="V.* ppa.* f p" postag_regexp="yes"></token>
+              </pattern>
+              <message>Vérifiez l’accord entre les pronoms « \1 » et « \3 ».</message>
+              <suggestion>toutes</suggestion>
```

Comment:
Alright, the FPs are gone! I propose the following change: let the `<marker>` include the `V ppa` as well. Then, suggest "toutes \4" AND `tous <match no="4" postag="(V.* ppa.*)(f p)" postag_replace="$1m p" postag_regexp="yes"/>` AND `toute <match no="4" postag="(V.* ppa.*)(f p)" postag_replace="$1f s" postag_regexp="yes"/>`.

Otherwise, this will happen:

"Ils étaient **tous** occupées à cacher se secret." → toutes

Our correction would make the sentence even more wrong.

And: since you made the rule very specific, we might as well propose the right form of "tout" and the right `V ppa`.
That will require subrules. One for \1 being `R pers suj 3 m p` (and the corresponding verb form), one for `R pers suj 3 m p` etc. And one rule for when we can't tell the gender (je, tu, nous, vous, ...)

Reply:
Some cases are already covered, this rule will scan for "tout" + V ppa matching "ils" or "elles".